ArduPlane/navigation.cpp

```diff
@@ -38,13 +40,28 @@ void Plane::loiter_angle_update(void)
     } else if (loiter.sum_cd == 0) {
         // use 1 cd for initial delta
         loiter_delta_cd = 1;
+        loiter.prev_lap_alt_cm = current_loc.alt;
+        loiter.next_sum_lap_cd = 36000;
     } else {
         loiter_delta_cd = target_bearing_cd - loiter.old_target_bearing_cd;
     }
     loiter.old_target_bearing_cd = target_bearing_cd;
     loiter_delta_cd = wrap_180_cd(loiter_delta_cd);
 
     loiter.sum_cd += loiter_delta_cd * loiter.direction;
+    loiter.reached_target_alt |= labs(current_loc.alt - next_WP_loc.alt) < 500;
```

Comment:
Do we really want this as a latching behavior or should it be flagging if we are currently at the target altitude? I'm inclined to say this shouldn't be latching and should just be an assign, then verify_loiter_to_alt manages the latching part (since it already is doing that as well).